File: Solutions/1695. Maximum Erasure Value/1695. Maximum Erasure Value.cpp
class Solution {
public:
    int maximumUniqueSubarray(vector<int>& nums) {
        int res=0,tmp=0,i=0,j=0;
        unordered_set<int> s;
        while(i<nums.size()&&j<nums.size()){
            if(s.find(nums[i])==s.end()){
                s.insert(nums[i]);
                tmp+=nums[i++];
                res=max(tmp,res);
            }else{
                s.erase(nums[j]);
                tmp-=nums[j++];
            }
        }
        return res;
    }
};

File: Solutions/1695. Maximum Erasure Value/1695. Maximum Erasure Value.go
func maximumUniqueSubarray(nums []int) int {
    i,j,tmp,res:=0,0,0,0
    set:=make(map[int]bool)
    for i<len(nums) && j<len(nums){
        if _,ok:=set[nums[i]];!ok{
            tmp+=nums[i]
            set[nums[i]]=true
            i+=1
            res=max(res,tmp)
        }else{
            delete(set,nums[j])
            tmp-=nums[j]
            j+=1
        }
    }
    return res
}

func max(a,b int) int{
    if a<b{
        return b
    }
    return a
}